port/zos390/omrvmem.c

```diff
@@ -1296,29 +1299,40 @@ omrvmem_default_large_page_size_ex(struct OMRPortLibrary *portLibrary, uintptr_t
 	}
 #endif /* OMR_ENV_DATA64 */
 
-	while (0 != PPG_vmem_pageSize[index]) {
-		if (0 != (OMRPORT_VMEM_MEMORY_MODE_EXECUTE & mode)) {
-			if ((ONE_M == PPG_vmem_pageSize[index]) &&
-				(0 != (OMRPORT_VMEM_PAGE_FLAG_PAGEABLE & PPG_vmem_pageFlags[index]))
-			) {
-				break;
+
+	/* Determine maximum pageable, and fixed large pages available. */
+	for (int i = 0; ((0 != PPG_vmem_pageSize[i]) && (ONE_M >= PPG_vmem_pageSize[i])); i++) {
+
+		if (ONE_M == PPG_vmem_pageSize[i]) {
+
+			if (0 != (OMRPORT_VMEM_PAGE_FLAG_PAGEABLE & PPG_vmem_pageFlags[i])) {
+				/* Pageable. Keep track of largest pageable size */
+				indexPageable = i;
 			}
 #if defined(OMR_ENV_DATA64)
-		} else {
-			if ((ONE_M == PPG_vmem_pageSize[index]) &&
-				(0 != (OMRPORT_VMEM_PAGE_FLAG_FIXED & PPG_vmem_pageFlags[index]))
-			) {
-				break;
+			else if (0 != (OMRPORT_VMEM_PAGE_FLAG_FIXED & PPG_vmem_pageFlags[i])) {
+				/* Fixed. Keep track of largest fixed size */
+				index = i;
 			}
-#endif
+#endif /* OMR_ENV_DATA64 */
 		}
-		index++;
 	}
+
+	/* 
+	 * When mode is OMRPORT_VMEM_MEMORY_EXECUTE we will only consider pagable large pages.
+	 * When flag is OMRPORT_VMEM_PAGE_FLAG_PAGEABLE_PREFERABLE, we prefer pageable unless fixed is larger.
+	 * NOTE: 32 bit only supports Pageable large pages. OMRPORT_VMEM_MEMORY_MODE_EXECUTE is guaranteed to be high during 32bit execution.
```

Comment:
I have no idea what you are trying to say in this comment. Please rephrase. Also for Z platforms we are usually refer to 31 bit platform

Reply:
Thanks Dmitri, this was remnants of the previous implementation.

Reply:
Done. See 923728f8c7f40a80bc81474ea4fca1b85c26ea21